### PR TITLE
Fix the logic for determining the commanded array source

### PR DESCRIPTION
**Pull request by bhazelton**
This fixes the logic around the commanded signal source:
- Use the proper table (`array_signal_source`) for the commanded source.
- Look up to 3 hours before the files start to get the commanded state because this is usually set well before the data taking begins. Also detect any changes in the commanded source during observing.
- Differentiate between antennas reporting as being in the correct state (using the reported FEM switch) and the observed antennas in the correct state. NOTE: this is supposed to add a line in the report, not sure if it is handled properly.

Still to do:
- [ ] fix logic for observed antennas in the correct state to account for digital noise

### CODE REVIEW

File: notebooks/rtp_summary.ipynb [hunk 1]
```diff
@@ -508,19 +508,60 @@
...
+    "        # still nothing, set it to None\n",
+    "        obs_source = None\n",
+    "    else:\n",
+    "        obs_source_times = []\n",
+    "        obs_source = []\n",
+    "        for obj in command_res:\n",
+    "            obs_source_times.append(obj.time)\n",
+    "            obs_source.append(obj.source)\n",
+    "\n",
+    "    if obs_source_time is not None:\n",
```
Comment (dstorer) [inline notebooks/rtp_summary.ipynb:544]: Should be `if obs_source is not None`?

File: notebooks/rtp_summary.ipynb [hunk 2]
```diff
@@ -508,19 +508,60 @@
...
+    "        # multiple commands\n",
+    "        times = []\n",
+    "        sources = []\n",
+    "        for obj in command_res:\n",
+    "            times.append(obj.time)\n",
+    "            sources.append(obj.source)\n",
+    "        inital_source = sources[np.argmax(times)]\n",
+    "    \n",
+    "    # check for any changes during observing\n",
+    "    command_res = session.get_array_signal_source(starttime=start_time, stoptime=start_time)\n",
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply notebooks/rtp_summary.ipynb:533]: So this logic was to get the last command before observing started. Then below this is a block that picks up any changes during observing. Not sure if this is the right architecture, but I do think we want to cleanly get the last one before observing starts.

[the remark this answers is not in the document]
Comment (bhazelton) [reply notebooks/rtp_summary.ipynb:533]: oh, oops. I was confused about what part of the code this comment was on. Yes, here it should be stop_time.

File: notebooks/rtp_summary.ipynb (hunk 1 above)
Comment (bhazelton) [reply notebooks/rtp_summary.ipynb:544]: yes, thanks!

**Review by dstorer — APPROVED** [review]

**Comment by jsdillon** [comment]
FYI @bhazelton @dstorer fixed a bug in this: https://github.com/HERA-Team/hera_notebook_templates/commit/c7e3123417779075a9a7cb9aa662605d69087e51

**Comment by bhazelton** [comment]
oh dear, sorry about that!

**Comment by jsdillon** [comment]
(re-running now to see if that was the only issue)

**Comment by jsdillon** [comment]
Found another bug: https://github.com/HERA-Team/hera_notebook_templates/commit/b319602c8e76df42618e6fedb6048672746c902f

**Comment by jsdillon** [comment]
And another: c21c8f72a7143f1ad2c3f90045dd3d06985ad08d

**Pull request merged**